src/containers/org/reducers/getOrganizationIntegrationDetailsReducer.js

```diff
@@ -31,10 +32,21 @@ export default function reducer(state :Map, action :SequenceAction) {
     },
     FAILURE: () => {
       if (state.hasIn([GET_ORGANIZATION_INTEGRATION_DETAILS, action.id])) {
-        return state.setIn([GET_ORGANIZATION_INTEGRATION_DETAILS, REQUEST_STATE], RequestStates.FAILURE);
+        return state
+          .setIn([GET_ORGANIZATION_INTEGRATION_DETAILS, ERROR], action.value)
+          .setIn([GET_ORGANIZATION_INTEGRATION_DETAILS, REQUEST_STATE], RequestStates.FAILURE);
       }
       return state;
     },
-    FINALLY: () => state.deleteIn([GET_ORGANIZATION_INTEGRATION_DETAILS, action.id]),
+    FINALLY: () => {
+      const storedAction = state.getIn([GET_ORGANIZATION_INTEGRATION_DETAILS, action.id]);
+      if (storedAction) {
+        const organizationId :UUID = storedAction.value;
+        return state
+          .setIn([INTEGRATION_DETAILS, organizationId], action.value)
+          .deleteIn([GET_ORGANIZATION_INTEGRATION_DETAILS, action.id]);
+      }
```

Comment:
Since you moved this to the `FINALLY` block, I don't think it needs to exist in the `SUCCESS` block.